pkg/policy/mapstate.go

```diff
@@ -766,8 +672,14 @@ func (ms *mapState) addKeyWithChanges(key Key, entry mapStateEntry, changes Chan
 // and 'deletes'.
 // The key is unconditionally deleted if 'owner' is nil, otherwise only the contribution of this
 // 'owner' is removed.
-func (ms *mapState) deleteKeyWithChanges(key Key, owner MapStateOwner, changes ChangeState) {
+// Note: key is only deleted if the 'precedence' matches with the precedence field of the found entry
+func (ms *mapState) deleteKeyWithChanges(owner MapStateOwner, key Key, precedence types.Precedence, changes ChangeState) {
 	if entry, exists := ms.get(key); exists {
+		if entry.Precedence != precedence {
+			fmt.Printf("SKIPPING DELETE DUE TO MISMATCHInG PRECEDENCE: %v != %v\n",
+				entry.Precedence, precedence)
```

Comment:
If the daemon needs to print messages, please use the logging library.

Reply:
Was leftover from local testing, removed!